From 2a9351d59ca2d1f79016ad79b4b69fc82dc14cd0 Mon Sep 17 00:00:00 2001
Message-ID: <2a9351d59ca2d1f79016ad79b4b69fc82dc14cd0.1740173176.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1740173176.git.sharlatanus@gmail.com>
References: <cover.1740173176.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 17 Feb 2025 14:05:39 +0000
Subject: [PATCH 04/51] gnu: libsep: Update to 1.4.0.

* gnu/packages/astronomy.scm (libsep): Update to 1.4.0.
[arguments] <test-target>: Set to "test".
<phases>: Add 'pre-check and use default 'check.

(python-sep) [arguments] <phases>: Add 'set-version.
[native-inputs]: Add python-setuptools-scm-next.

Change-Id: I04ee0704fb52c1fb02d6efa0a2bc6b7e07a516de
---
 gnu/packages/astronomy.scm | 42 ++++++++++++++++++++++++--------------
 1 file changed, 27 insertions(+), 15 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index e82a9274dfe..25826c22703 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -1245,7 +1245,7 @@ (define-public libpasastro
 (define-public libsep
   (package
     (name "libsep")
-    (version "1.2.1")
+    (version "1.4.0")
     (source
      (origin
        (method git-fetch)
@@ -1254,20 +1254,21 @@ (define-public libsep
              (commit (string-append "v" version))))
        (file-name (git-file-name name version))
        (sha256
-        (base32 "0sag96am6r1ffh9860yq40js874362v3132ahlm6sq7padczkicf"))))
+        (base32 "03a9v3g8bq5sqq0ckxzkk63vij2y6ljpmymdvvvvb72q175pzpkd"))))
     (build-system cmake-build-system)
     (arguments
      (list
-      #:make-flags #~(list (string-append "CC=" #$(cc-for-target))
-                           (string-append "PREFIX=" #$output))
-      #:phases #~(modify-phases %standard-phases
-                   (replace 'check
-                     (lambda* (#:key tests? #:allow-other-keys)
-                       (when tests?
-                         (chdir "../source")
-                         (invoke "make"
-                                 (string-append "CC=" #$(cc-for-target))
-                                 "test")))))))
+      #:make-flags
+      #~(list (string-append "CC=" #$(cc-for-target))
+              (string-append "PREFIX=" #$output))
+      #:test-target "test"
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'check 'pre-check
+            (lambda* (#:key tests? #:allow-other-keys)
+              (when tests?
+                (chdir "../source")
+                (setenv "CC" #$(cc-for-target))))))))
     (native-inputs
      (list python-wrapper))
     (home-page "https://github.com/kbarbary/sep")
@@ -6137,11 +6138,22 @@ (define-public python-sep
     (name "python-sep")
     (build-system pyproject-build-system)
     (arguments
-     (list #:test-flags #~(list "test.py")))
+     (list
+      #:test-flags #~(list "test.py")
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-after 'unpack 'set-version
+            (lambda _
+              (setenv "SETUPTOOLS_SCM_PRETEND_VERSION"
+                      #$(package-version this-package)))))))
     (native-inputs
-     (list python-cython python-pytest python-setuptools python-wheel))
+     (list python-cython
+           python-pytest
+           python-setuptools
+           python-setuptools-scm-next
+           python-wheel))
     (propagated-inputs
-     (list  python-numpy))
+     (list python-numpy))
     (synopsis "Python library for Source Extraction and Photometry")))
 
 (define-public python-sep-pjw
-- 
2.47.1

